### ISSUE TITLE
please provide man page; use POD documentation

### ISSUE THREAD

**Issue by onlyjob** [so-fancy/diff-so-fancy#195]
Please introduce man page and POD documentation.

Man page can be conveniently generated from POD markup using [Pod::Usage](http://perldoc.perl.org/Pod/Usage.html) core module.

Thanks.

**Comment by stevemao** [comment]
I like this. PR welcome :)

**Comment by scottchiefbaker** [comment]
I think this is a great idea. We'd definitely welcome either PRs, or feedback on exactly what should be covered in the manpage. I've never used POD or created a manpage.

**Comment by Alok** [comment]
You could use Pandoc to convert the README into a manpage, or at least write a manpage in markdown rather than `groff`.

**Comment by scottchiefbaker** [comment]
I don't have expertise in this area, so I'm willing to accept PRs for this.

**Comment by petdance** [comment]
I do all the docs for `ack` in POD and convert them to manpages.  I'd be glad to help.  https://github.com/beyondgrep/ack3

**Comment by scottchiefbaker** [comment]
@petdance fantastic. If you're up to writing some POD documentation we'll gladly review it and get it in to the repo.

**Comment by petdance** [comment]
I don’t know enough about the tool to write docs. I can help with the POD to manual and HTML conversion.

**Comment by scottchiefbaker** [comment]
@petdance I've never written POD before. Where is a good starting point?

**Comment by petdance** [comment]
`$ perldoc perlpod` to see the man page for POD.  It has all the details.

POD is like Markdown, just older.  You do things like `=head1` for level headings, `B<foo>` to make things bold, and so on.  For an example, see https://github.com/beyondgrep/ack3/blob/dev/lib/App/Ack.pm

Paragraph-level directives start with `=` and are in the leftmost column.  Inline markup is always 'x<whatever>' where `x` is a capital letter.

**Comment by scottchiefbaker** [comment]
Is the whole point of POD just to output man pages?

**Comment by petdance** [comment]
No, it's also for internal documentation, but I think in this case you're just looking for man pages.

Another benefit of using POD, since you're using Perl anyway, is that there is such a rich set of tools in place to process POD.  So you can create help on the fly with Pod::Usage, or you can create man pages with `pod2man`, or you can create HTML with `pod2html`, like here: https://beyondgrep.com/documentation/ack-v3.3.1-man.html

**Comment by matthewpersico** [comment]
[Pod::Usage](https://perldoc.perl.org/Pod/Usage.html)

**Comment by buhtz** [comment]
> I do all the docs for `ack` in POD and convert them to manpages. I'd be glad to help. https://github.com/beyondgrep/ack3

Which command do you use on the shell to convert a pod into a man page? I couldn't find an indication for this in your repo and also not in the pod docu. I might miss something.

**Comment by scottchiefbaker** [comment]
@petdance are you still willing to help us write some POD? I think the basics in `diff-so-fancy --help` would be sufficient.

**Comment by petdance** [comment]
I'm glad to help with formatting. I don't know enough about the tool to write the content.

**Comment by scottchiefbaker** [comment]
Give a look at the POD I wrote in [728de7b](https://github.com/so-fancy/diff-so-fancy/commit/728de7b1937a222e603388960519f300a3de31c0). It's good enough to generate a man page. I'm going to consider this issue closed. If you think we need more please re-open.

**Comment by petdance** [comment]
Unrelated I updated the page at https://altbox.dev/tool/diff-so-fancy
